redisvl/llmcache/semantic.py

```diff
@@ -74,25 +121,40 @@ def check(
         num_results: int = 1,
         fields: List[str] = ["response"],
     ) -> Optional[List[str]]:
-        """Checks whether the cache contains the specified key."""
+        """Checks whether the cache contains the specified prompt or vector.
+
+        Args:
+            prompt (Optional[str], optional): The prompt to check. Defaults to None.
+            vector (Optional[List[float]], optional): The vector to check. Defaults to None.
+            num_results (int, optional): The number of results to return. Defaults to 1.
+            fields (List[str], optional): The fields to return. Defaults to ["response"].
+
+        Raises:
+            ValueError: If neither prompt nor vector is specified.
+
+        Returns:
+            Optional[List[str]]: The response(s) if the cache contains the prompt or vector.
+        """
         if not prompt and not vector:
             raise ValueError("Either prompt or vector must be specified.")
 
-        query = create_vector_query(
-            return_fields=fields,
+        if not vector:
+            vector = self._provider.embed(prompt)  # type: ignore
+
+        v = VectorQuery(
```

Comment:
I'd suggest we use a range query for this use case - less client side processing. If you agree -- we will want to flip the polarity of the threshold such that smaller is more "strict"

Reply:
premature Optimization is the root of all evil. Next release.